controllers/managedfusionoffering_controller.go

```diff
@@ -85,13 +97,118 @@ func pluginReconcile(r *ManagedFusionOfferingReconciler) (ctrl.Result, error) {
+		if err := r.own(storageCluster); err != nil {
+			return err
+		}
+		sizeAsString := r.managedFusionOffering.Spec.Config["usableCapacityInTiB"]
+
+		// Setting hardcoded value here to force no MCG deployment
+		enableMCGAsString := "false"
+		if enableMCGRaw, exists := r.managedFusionOffering.Spec.Config[enableMCGKey]; exists {
+			enableMCGAsString = enableMCGRaw
+		}
+		r.Log.Info("Requested add-on settings", usableCapacityInTiBKey, sizeAsString, enableMCGKey, enableMCGAsString)
+		desiredSize, err := strconv.Atoi(sizeAsString)
+		if err != nil {
+			return fmt.Errorf("Invalid storage cluster size value: %v", sizeAsString)
+		}
+
+		// Convert the desired size to the device set count based on the underlaying OSD size
+		desiredDeviceSetCount := int(math.Ceil(float64(desiredSize) / templates.ProviderOSDSizeInTiB))
+
+		// Get the storage device set count of the current storage cluster
+		currDeviceSetCount := 0
+		if desiredStorageDeviceSet := findStorageDeviceSet(storageCluster.Spec.StorageDeviceSets, deviceSetName); desiredStorageDeviceSet != nil {
+			currDeviceSetCount = desiredStorageDeviceSet.Count
+		}
+
+		// Get the desired storage device set from storage cluster template
+		sc := templates.ProviderStorageClusterTemplate.DeepCopy()
+		var ds *ocsv1.StorageDeviceSet = nil
+		if desiredStorageDeviceSet := findStorageDeviceSet(sc.Spec.StorageDeviceSets, deviceSetName); desiredStorageDeviceSet != nil {
+			ds = desiredStorageDeviceSet
+		}
+
+		// Prevent downscaling by comparing count from secret and count from storage cluster
+		setDeviceSetCount(r, ds, desiredDeviceSetCount, currDeviceSetCount)
+
+		// Check and enable MCG in Storage Cluster spec
+		mcgEnable, err := strconv.ParseBool(enableMCGAsString)
+		if err != nil {
+			return fmt.Errorf("Invalid Enable MCG value: %v", enableMCGAsString)
+		}
+		if err := ensureMCGDeployment(r, sc, mcgEnable); err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Unable to create/update StorageCluster: %v", err)
+
+	}
+
+	return nil
+}
+
+func findStorageDeviceSet(storageDeviceSets []ocsv1.StorageDeviceSet, deviceSetName string) *ocsv1.StorageDeviceSet {
+	for index := range storageDeviceSets {
+		item := &storageDeviceSets[index]
+		if item.Name == deviceSetName {
+			return item
+		}
+	}
+	return nil
+}
+
+func setDeviceSetCount(r *ManagedFusionOfferingReconciler, deviceSet *ocsv1.StorageDeviceSet, desiredDeviceSetCount int, currDeviceSetCount int) {
+	r.Log.Info("Setting storage device set count", "Current", currDeviceSetCount, "New", desiredDeviceSetCount)
+	if currDeviceSetCount <= desiredDeviceSetCount {
+		deviceSet.Count = desiredDeviceSetCount
+	} else {
+		r.Log.V(-1).Info("Requested storage device set count will result in downscaling, which is not supported. Skipping")
+		deviceSet.Count = currDeviceSetCount
+	}
+}
+
+func ensureMCGDeployment(r *ManagedFusionOfferingReconciler, storageCluster *ocsv1.StorageCluster, mcgEnable bool) error {
```

Comment:
MCG is not required in DF

Reply:
I think there is no harm in keeping MCG, anyway we are not enabling it.

Reply:
ack